invoice.py

```diff
@@ -222,23 +243,43 @@ def generate_facturae_wizard(cls, invoices):
         pass
 
     @classmethod
-    def generate_facturae_default(cls, invoices, certificate_password):
+    def generate_facturae(cls, invoices, certificate=None, service=None):
+        Configuration = Pool().get('account.configuration')
+
+        config = Configuration(1)
+        transaction = Transaction()
+        context = transaction.context
+
         to_write = ([],)
         for invoice in invoices:
             if invoice.invoice_facturae:
                 continue
             facturae_content = invoice.get_facturae()
             invoice._validate_facturae(facturae_content)
             if backend.name != 'sqlite':
-                invoice_facturae = invoice._sign_facturae(
-                    facturae_content, certificate_password)
+                invoice_facturae = invoice._sign_facturae(facturae_content,
```

Comment:
UserErrors should be captured if we're in a worker method. Maybe you need a wrapper around generate_facturae for the worker.

You may prefer to use an instance method to simplify everything.